ui/components/CustomActions.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from "react";
+import { Link } from "..";
+import {
+  Bucket,
+  Kustomization,
+  HelmRelease,
+  HelmRepository,
+  GitRepository,
+  OCIRepository,
+  HelmChart,
+} from "../lib/objects";
+import Button from "./Button";
+import Spacer from "./Spacer";
+
+type Props = {
+  resource:
+    | Bucket
+    | Kustomization
+    | HelmRelease
+    | HelmRepository
+    | GitRepository
+    | OCIRepository
+    | HelmChart;
+};
+
+export const EditButton = ({ resource }: Props) => {
+  const hasCreateRequestAnnotation =
+    resource.obj.metadata.annotations?.["templates.weave.works/create-request"];
+
+  return hasCreateRequestAnnotation ? (
+    <Link to={`/resources/${resource.name}/edit`}>
+      <Button>Edit</Button>
+    </Link>
+  ) : null;
+};
+
+const CustomActions = ({ actions }) => {
```

Comment:
⚠️ same as above - if you could add a defined props type here that'd be awesome!